packages/ubuntu_flavor/lib/src/ubuntu_flavor.dart

```diff
@@ -1,100 +1,34 @@
 import 'package:meta/meta.dart';
 import 'package:platform_linux/platform.dart';
-
 import 'package:ubuntu_flavor/src/ubuntu_flavor_stub.dart'
     if (dart.library.io) 'ubuntu_flavor_io.dart';
 
-@immutable
-class UbuntuFlavor {
-  const UbuntuFlavor({
-    required this.id,
-    required this.name,
-  });
-
-  static const budgie = UbuntuFlavor(
-    id: 'ubuntu-budgie',
-    name: 'Ubuntu Budgie',
-  );
-
-  static const cinnamon = UbuntuFlavor(
-    id: 'ubuntucinnamon',
-    name: 'Ubuntu Cinnamon',
-  );
-
-  static const edubuntu = UbuntuFlavor(
-    id: 'edubuntu',
-    name: 'Edubuntu',
-  );
-
-  static const kubuntu = UbuntuFlavor(
-    id: 'kubuntu',
-    name: 'Kubuntu',
-  );
-
-  static const kylin = UbuntuFlavor(
-    id: 'ubuntukylin',
-    name: 'Ubuntu Kylin',
-  );
-
-  static const lubuntu = UbuntuFlavor(
-    id: 'lubuntu',
-    name: 'Lubuntu',
-  );
-
-  static const mate = UbuntuFlavor(
-    id: 'ubuntu-mate',
-    name: 'Ubuntu MATE',
-  );
-
-  static const studio = UbuntuFlavor(
-    id: 'ubuntustudio',
-    name: 'Ubuntu Studio',
-  );
-
-  static const ubuntu = UbuntuFlavor(
-    id: 'ubuntu',
-    name: 'Ubuntu',
-  );
-
-  static const unity = UbuntuFlavor(
-    id: 'ubuntu-unity',
-    name: 'Ubuntu Unity',
-  );
-
-  static const xubuntu = UbuntuFlavor(
-    id: 'xubuntu',
-    name: 'Xubuntu',
-  );
-
-  static UbuntuFlavor? detect([
+enum UbuntuFlavor {
+  budgie('Ubuntu Budgie'),
+  cinnamon('Ubuntu Cinnamon'),
+  edubuntu('Edubuntu'),
+  kubuntu('Kubuntu'),
+  kylin('Ubuntu Kylin'),
+  lubuntu('Lubuntu'),
+  mate('Ubuntu MATE'),
+  studio('Ubuntu Studio'),
+  ubuntu('Ubuntu'),
+  unity('Ubuntu Unity'),
+  xubuntu('Xubuntu'),
+  unknown('Unknown');
+
+  const UbuntuFlavor(this.name);
+  factory UbuntuFlavor.detect([
     @visibleForTesting Platform platform = const LocalPlatform(),
   ]) {
     return detectUbuntuFlavor(platform);
   }
 
-  final String id;
-  final String name;
-
-  UbuntuFlavor copyWith({
-    String? id,
-    String? name,
-  }) {
-    return UbuntuFlavor(
-      id: id ?? this.id,
-      name: name ?? this.name,
-    );
-  }
-
-  @override
-  String toString() => 'UbuntuFlavor(id: $id, name: $name)';
-
-  @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) return true;
+  factory UbuntuFlavor.fromId(String id) => values.firstWhere(
+        (flavor) => flavor.id == id,
+        orElse: () => unknown,
+      );
 
-    return other is UbuntuFlavor && other.id == id && other.name == name;
-  }
-
-  @override
-  int get hashCode => Object.hash(id, name);
+  final String name;
```

Comment:
Name is a bit problematic as a field in enums since it exists here:
https://api.flutter.dev/flutter/dart-core/EnumName/name.html

So it's better to come up with something else.

Reply:
I think that `Enum.name` might rely on `toString`, so you might be able to override `toString` instead and point it to a field named something else and the already existing `name` getter would work as we want. 🤔

Reply:
Damn, doesn't seem like it does anymore.

Reply:
> the result of MyEnum.value1.name is the string "value1"

so I guess that's pretty clear and we shouldn't modify that behavior. I'd suggest using `name` and `displayName` instead of `id` and `name` then, even if this leads to some annoyances

Reply:
Yeah, sounds good. Then those dirty toString splits can be removed too, I probably forgot about `name` when I wrote those.